src/eko/output/manipulate.py

```diff
@@ -134,12 +138,19 @@ def flavor_reshape(
         ops = elem.operator
         errs = elem.error
         if targetpids is not None and inputpids is None:
+            # update metadata
+            eko.update_metadata({"rotations": {"targetpids": targetpids}})
             ops = np.einsum("ca,ajbk->cjbk", targetpids, ops)
             errs = np.einsum("ca,ajbk->cjbk", targetpids, errs)
         elif inputpids is not None and targetpids is None:
+            # update metadata
+            eko.update_metadata({"rotations": {"inputpids": inputpids}})
             ops = np.einsum("ajbk,bd->ajdk", ops, inv_inputpids)
             errs = np.einsum("ajbk,bd->ajdk", errs, inv_inputpids)
         else:
+            # update metadata
+            eko.update_metadata({"rotations": {"inputpids": inputpids}})
```

Comment:
the call should move down to were the write is happening (else we get out of sync) .. note that we're casting to list (see [comment below](https://github.com/NNPDF/eko/pull/162/files/99ee17fdfa0cd5c3565442aee2d6323c9d339b1b..a85fd194824bc4f195a4250682dca0b18acdec3f#r1021860266) )

Reply:
This made sense: we want operations to be as atomic as possible (in the ACID sense)